### ISSUE TITLE
remix-dev/cli loadEnv should be before readConfig(), In order to use environment variables in remix.config.js

### ISSUE THREAD

**Issue by jackyleefu** [remix-run/remix#2418]
### What version of Remix are you using?

1.3.2

### Steps to Reproduce

233: let config = await readConfig(remixRoot);
...
236: await loadEnv(config.rootDirectory);

### Expected Behavior

236: await loadEnv(config.rootDirectory);
...
233: let config = await readConfig(remixRoot);

### Actual Behavior

233: let config = await readConfig(remixRoot);
...
236: await loadEnv(config.rootDirectory);

**Comment by sergiodxa** [comment]
If this is changed. Then the behavior is Remix.config.js would change based on the server, on Remix App Server it will get env variable from .env but on the rest it will not.

I think it’s better to keep it on the current way IMO.

**Comment by kiliman** [comment]
I agree. Not sure why you would need to define environment variables in *remix.config*. There are many ways to do this already: *.env*, `cross-env`, platform UI (set for production), etc.

**Comment by jackyleefu** [comment]
Because I need to load custom routes from database in remix.config.js, and database connection string is in environment variable, although I can use cross-env, it has to be like this:

dev: `cross-env DOTENV_CONFIG_PATH=.env.local node -r dotenv/config node_modules/@remix-run/dev/cli.js dev`

build: `node -r dotenv/config node_modules/@remix-run/dev/cli.js build`

**Comment by jackyleefu** [comment]
> I agree. Not sure why you would need to define environment variables in _remix.config_. There are many ways to do this already: _.env_, `cross-env`, platform UI (set for production), etc.

**Comment by kiliman** [comment]
> You can call dotenv yourself in remix.config.js

One of the benefits of using actual JS files for config instead of JSON.

**Comment by machour** [comment]
Converting this to a Proposal discussion, as per our [Development Process](https://remix.run/docs/en/v1/pages/contributing#development-process).
